### PR TITLE
Set pyopenms numpy install<2.0

### PR DESCRIPTION
**Pull request by matteopilz**
## Checklist
- [ ] Make sure that you are listed in the AUTHORS file
- [ ] Add relevant changes and new features to the CHANGELOG file
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [ ] Updated or added python bindings for changed or new classes (Tick if no updates were necessary.)

### How can I get additional information on failed tests during CI
<details>
  <summary>Click to expand</summary>
If your PR is failing you can check out

- The details of the action statuses at the end of the PR or the "Checks" tab.
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. Use the "Show filters" capability on the top right to search for your PR number.
  If you click in the column that lists the failed tests you will get detailed error messages.

</details>

### Advanced commands (admins / reviewer only)
<details>
  <summary>Click to expand</summary>
  
- `/reformat` (experimental) applies the clang-format style changes as additional commit. Note: your branch must have a different name (e.g., yourrepo:feature/XYZ) than the receiving branch (e.g., OpenMS:develop). Otherwise, reformat fails to push.
- setting the label "NoJenkins" will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)
- commenting with `rebuild jenkins` will retrigger Jenkins-based CI builds
  
</details>

---
:warning: Note: Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

___

### **PR Type**
enhancement, configuration changes

___

### **Description**
- Restricted `numpy` version to be less than 2.0 in the `setup.py` file.
- Updated CI workflows to install `numpy` version less than 2.0.
- Renamed the test workflow and updated its trigger workflow.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Configuration changes
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>setup.py</strong><dd><code>Restrict `numpy` version to below 2.0 in setup configuration.</code></dd></summary>
<hr>
      
src/pyOpenMS/setup.py

- Set `numpy` version to be less than 2.0 in `install_requires`.

</details>
    

  </td>
  <td><a href="https://github.com/OpenMS/OpenMS/pull/7515/files#diff-1343deca77d77a5eb66c3fddd2f25c883ba408f22d13c33b760b691b5ea82d75">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>pyopenms-wheels.yml</strong><dd><code>Restrict `numpy` version to below 2.0 in CI workflow.</code>&nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
.github/workflows/pyopenms-wheels.yml

<li>Set <code>numpy</code> version to be less than 2.0 in multiple <code>pip install</code> <br>commands.<br>

</details>
    

  </td>
  <td><a href="https://github.com/OpenMS/OpenMS/pull/7515/files#diff-084f20ae9727db0905c5fbbc32ef797e048c87887ccfc57b0f89b2bf82f45979">+4/-4</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>test_pyopenms.yml</strong><dd><code>Update workflow name and trigger in CI configuration.</code>&nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
.github/workflows/test_pyopenms.yml

<li>Rename workflow to <code>Test pyOpenMS on different platforms</code>.<br> <li> Update trigger workflow to <code>pyopenms-wheels-and-packages</code>.<br>

</details>
    

  </td>
  <td><a href="https://github.com/OpenMS/OpenMS/pull/7515/files#diff-770c7205b412d193b98c06200794e25724f74f96dcfd5b6a6a3726d6ea01cdea">+3/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: .github/workflows/test_pyopenms.yml [hunk 1]
```diff
@@ -1,9 +1,9 @@
-name: Test nightly pyOpenMS on different platforms
+name: Test pyOpenMS on different platforms
 
 on:
   workflow_dispatch: # Trigger manually
-  workflow_run:
-    workflows: [ pyopenms-wheels ]
+  workflow_run: # Trigger on pyoepnems-wheels
```
Comment (timosachsenberg) [inline .github/workflows/test_pyopenms.yml]:
```suggestion
  workflow_run: # Trigger on pyopenems-wheels
```

**Comment by timosachsenberg** [comment]
looks good to me. guess the quotes " are fine

File: .github/workflows/pyopenms-wheels.yml [hunk 2]
```diff
@@ -103,7 +103,7 @@ jobs:
           pip install -U pip
           pip install -U autowrap
           pip install -U pytest
-          pip install -U numpy
+          pip install numpy<2.0
```
Comment (mr-c) [inline .github/workflows/pyopenms-wheels.yml:106]: all of these `<` need to be inside quotes

Comment (mr-c) [reply .github/workflows/pyopenms-wheels.yml:106]: Ah, I see https://github.com/OpenMS/OpenMS/commit/7e751ffc51507a6e12b5eedccb05a1ff52e0561e

**Pull request merged**